fabric-events-interaction-v0/src/main/java/net/fabricmc/fabric/mixin/event/interaction/MixinClientPlayerInteractionManager.java

```diff
@@ -141,4 +146,9 @@ public void interactEntityAtLocation(PlayerEntity player, Entity entity, EntityH
 			return;
 		}
 	}
+
+	@Inject(at = @At(value = "INVOKE", target = "Lnet/minecraft/block/Block;onBroken(Lnet/minecraft/world/WorldAccess;Lnet/minecraft/util/math/BlockPos;Lnet/minecraft/block/BlockState;)V"), method = "breakBlock", locals = LocalCapture.CAPTURE_FAILHARD)
+	private void onBlockBroken(BlockPos pos, CallbackInfoReturnable<Boolean> cir, World world, BlockState state, Block block, FluidState fluidState, boolean bl) {
```

Comment:
will it be weird if you run post block break in the client but not pre block break? I would just not call it in the client or make a new callback especially for the client since you are calling World#getBlockEntity

Reply:
Client/Server distinction isn't a big issue since we have a world.

Reply:
what I meant is that it will get confusing how there is only the after block break for the client, there isn't before block break for the client